TrafficSimulator/Assets/Car/AutoDrive.cs

```diff
@@ -147,6 +157,40 @@ void Update()
             {
                 P_MoveToNextPosition();
             }
+            updateOccupiedNodes();
+        }
+
+        // Update the list of nodes that the vehicle is currently occupying
+        private void updateOccupiedNodes()
+        {
+            foreach (LaneNode node in _occupiedNodes)
+            {
+                node.UnsetVehicle(_vehicle);
+            }
+            _occupiedNodes.Clear();
+            _occupiedNodes = getOccupiedNodes(_currentNode);
+            foreach (LaneNode node in _occupiedNodes)
+            {
+                node.SetVehicle(_vehicle);
+            }
+        }
+
+        // Get the list of nodes that the vehicle is currently occupying by moving backwards from the current position until out of vehicle bounds
+        private List<LaneNode> getOccupiedNodes(LaneNode node)
```

Comment:
The function takes a LaneNode as parameter which is unneccessary, and then it is only used for looking backwards. Forwards is based on `currentNode`. The parameter should be removed and the internal logic should be based on the `currentNode` instead, checking forward and then checking backwards from there